### ISSUE TITLE
There is no way to create a new database or directly restore backup file as new database

### ISSUE THREAD

**Issue by peterlukerow** [microsoft/azuredatastudio#6997]
Version: 1.10.0 (system setup)
Commit: 5fdb967ffcea41df0335ec17cfa396a694bb174e
Date: 2019-08-14T18:28:10.311Z
VS Code 1.37.0
Electron: 4.2.7
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362

Issue number 1 - Steps to Reproduce:

1. In server explorer pane, right click "Databases" node.
2. PROBLEM: There is no create database or restore database option on the context menu.

Issue number 2 - Steps to Reproduce:
1. In server explorer pane, right click the server and click "Manage" from context menu
2. On server dashboard click the "Restore" option in the tasks section.
3. Select a bak file and type in a new database name in the Target Database field.
4. PROBLEM: The confirmation button is disabled so you can't restore it. Further problem there is no option in the tasks section to create a database. Thus there is literally no UI for creating a database!!

I really, really want to use Azure Data Studio over SSMS but you make life so very difficult sometimes. Often you do things a lot better than SSMS but then you have complete junk like the above where trivially basic database server management 101 features are missing. Then in response rather than going to the drawing board and saying what are the basic features that people coming from SSMS would want you request that we write out each individual item as a separate request. This is insane!! Yes I could probably write some T-SQL script to do this, but this GUI application and therefore basic functionality like creating a database and restoring from a BAK file should be available in the GUI and not force the use of T-SQL all the time.

**Comment by udeeshagautam** [comment]
@peterlukerow  thanks for reporting this issue. 
For the issue 1 - Currently Backup/Restore context menu is available on the individual Database node. Restore is also available from Server Dashboard.

For issue 2 Can you please elaborate on : "The confirmation button is disabled"
I tried reproing the same with following steps
 - Open the restore dialog from Server --> Manage -->Restore
 -  Select an Existing .bak file created from some other DB on the server
 - Type in new db name (not previously existing) in the Target Database Editable combo box.
 - With this both "Script" and "Restore" buttons are enabled and I am able to restore. 

Can you please let me know if I am missing something in repro steps.

![image](https://user-images.githubusercontent.com/46980425/64991275-6c8e6780-d886-11e9-9606-7907e7164413.png)

**Comment by peterlukerow** [comment]
I've been on holiday for a week, sorry for the later reply. Upon my return a new version of ADS was available and now issue 2 is no longer reproducible. When some one says "the confirmation button" in reference to UI in the current context is meant to say the button that confirms the action, i.e. when you click it the action is carried out. In future I will make sure to be explict.

You've not understood the problem with issue 1. To put it simply:
There is 100% zero ways to create a brand new database other in the UI other than via the now fixed method that was previously a problem as described in issue 2.

![issue1-server-manage-no-create-option](https://user-images.githubusercontent.com/28951140/65135058-36272880-d9fd-11e9-948a-d98ebd8cc564.png)
As in the above image - the manage server dashboard gives you no options whatsoever to create a new empty database.

![issue1-context-menu-no-create-database-or-restore-option](https://user-images.githubusercontent.com/28951140/65135059-36272880-d9fd-11e9-818e-03f9487e41cc.png)
As in the above image - after right clicking on the "Databases" node in the server explorer pane the only option on the context menu is refresh, there is no create database or restore database option. Both options should be there. The reason for restore being here (as it is in SSMS) is so you can restore a .bak file to a brand new database that doesn't yet exist.

I hope the above screenshots and information clears up issue 1.
Issue 2 is now fixed in this version of ADS:
Version: 1.11.0 (system setup)
Commit: 460c739a8d90bdbbde39e01abfb8ecdc278ea787
Date: 2019-09-06T18:08:50.974Z
VS Code 1.37.0
Electron: 4.2.9
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by udeeshagautam** [comment]
Thanks for the explanation. Apologies for not understating the issue fully the first time. Create database is not an option present in ADS at this point. Regarding restore - we will work towards the restore option being available from the databases context menu.

**Comment by peterlukerow** [comment]
Is this getting any closer to the getting-done stage?

Yet again had the need to create a new database, and yet again even with the latest available (not beta) version I still cannot create a new database and thus am forced to use SSMS again.

**Comment by llali** [comment]
@peterlukerow the "New Database" menu item in the object explorer is available by installing "New Database" extension from the market place:
![image](https://user-images.githubusercontent.com/17187348/76129699-eb09cd80-5fbc-11ea-8198-c72dc150bb7f.png)
 
There's also one for "Delete Database". The idea is that the missing features which not every user need will be available through the extensions. So user can decide to add them or not.

**Comment by llali** [comment]
I verified the restore functionalities work fine. If I'm missing something please let me know. There's not any "Restore Database" option in the menu item and bu as mentioned above that can be added through an extension.

**Comment by peterlukerow** [comment]
@llali 
I'm sorry but basic operations like new database and restore database are **NOT** things that should require an extension they should be built into the product.

Please stop ignoring users! When I search open issues there lots and lots of issues that reporting problems with functionality that any one coming from SSMS expects to be there. Please, please reconsider these product choices you're making, 😥😥😥

I really want the day to come where I no longer have to have SSMS open at the same time as Azure Data Studio.

I suggest if possible you get a group of SSMS users together who have been using ADS for a while and ask them about the things that are missing or not as good and act on those as soon as possible. Why maintain 2 products that for the most part do the same thing?

**Comment by peterlukerow** [comment]
@llali 
Even worse the new database extension sends me to a github website and makes me download and install manually, it's not actually even a proper part of the extensions marketplace, i.e. one that you just click install and wait until it's done.

**Comment by llali** [comment]
@peterlukerow I'm sorry for the inconvenience. The UI features in ADS are in general different than SSMS. We have the dashboard in ADS which includes the actions to do on the database and database objects. SSMS on the other hand, has the actions in the Object Explorer tree. It's understandable that SSMS users are used to have actions in the tree but the problem of too many menu items which make it hard to discover made us realize the dashboard might be a better option to visualize the actions. But we still let user create the actions they need to the tree through the extensibility model in ADS.  The extensions in ADS are provided by the community and not all of them are hosted by us. That's why for some of them user still need to download the vsix from somewhere else. I tested the extension I mentioned myself and verified the actions work correctly.

**Comment by peterlukerow** [comment]
@llali 
That would be fine *EXCEPT* the ADS dashboard does *not* have a new database option!

It would be fine if the functionality existed in different places but the simple fact is it doesn't!

Stating that you need to add multiple extensions for basic things like "new database" is quite simply unacceptable! Also that particular extension is written by a Microsoft person unless I'm misunderstanding what Github said.

**Comment by llali** [comment]
@peterlukerow you're right and I can review this with the team to consider adding main actions to the dashboard. But there's also other ways provided in ADS  to do the main actions like this. Code snippet in ADS has support for common TSQLs. You can type "create" in an editor to see the options. It generates this script for "create database"
-- Create a new database called 'DatabaseName'
-- Connect to the 'master' database to run this snippet
USE master
GO
-- Create the new database if it does not exist already
IF NOT EXISTS (
    SELECT [name]
        FROM sys.databases
        WHERE [name] = N'DatabaseName'
)
CREATE DATABASE DatabaseName
GO

**Comment by peterlukerow** [comment]
@llali 
I'm sure ADS does have code snippets but MS has to realise that people work in different ways and force people to use 1 specific method rather than having the choice like SSMS.

Also that snippet is very basic and doesn't take into account of having different locations for data and log files, nor auto growth and size restrictions.

The whole point of a GUI is to allow things to be done quickly without having to look up, in this case, SQL commands and all the various options needed to do what they want.

I have installed the extensions you mentioned and it looks like they suffer the same issue as the snippet, i.e. all it does it ask for a name, uber sigh!
